swap/src/env.rs

```diff
@@ -107,6 +92,33 @@ fn sync_interval(avg_block_time: Duration) -> Duration {
     max(avg_block_time / 10, Duration::from_secs(1))
 }
 
+pub fn new(is_testnet: bool, asb_config: &asb::config::Config) -> Config {
+    let env_config = if is_testnet {
+        Testnet::get_config()
+    } else {
+        Mainnet::get_config()
+    };
+
+    let env_config =
+        if let Some(bitcoin_finality_confirmations) = asb_config.bitcoin.finality_confirmations {
+            Config {
+                bitcoin_finality_confirmations,
+                ..env_config
+            }
+        } else {
+            env_config
+        };
+
+    if let Some(monero_finality_confirmations) = asb_config.monero.finality_confirmations {
+        Config {
+            monero_finality_confirmations,
+            ..env_config
+        }
```

Comment:
📯 
Am I misreading this or is this part wrong? 

If `asb_config.bitcoin.finality_confirmations` is set then you only set the `bitcoin_finality_confirmations` and default for `monero_finality_confirmations`.
 and if it `asb_config.bitcoin.finality_confirmations` none you will set `monero_finality_confirmations`... 

I guess you want to have both set if they are `Some` right?

Reply:
All good, I was confused about the indentations.